packages/admin/src/pages/scheme/[schemeId]/manage-editors/add-editor.page.tsx

```diff
@@ -0,0 +1,110 @@
+import { Button, FlexibleQuestionPageLayout, TextInput } from 'gap-web-ui';
+import CustomLink from '../../../../components/custom-link/CustomLink';
+import Meta from '../../../../components/layout/Meta';
+import InferProps from '../../../../types/InferProps';
+import { GetServerSidePropsContext } from 'next';
+import {
+  getSessionIdFromCookies,
+  getUserTokenFromCookies,
+} from '../../../../utils/session';
+import QuestionPageGetServerSideProps from '../../../../utils/QuestionPageGetServerSideProps';
+import { addSchemeEditor } from '../../../../services/SchemeService';
+
+type PageBodyResponse = {
+  editorEmailAddress: string;
+};
+
+export async function getServerSideProps(context: GetServerSidePropsContext) {
+  const { schemeId } = context.params as Record<string, string>;
+  const schemeName = decodeURIComponent(context.query.schemeName as string);
+
+  const newEmailAddress = context.query.newEmailAddress
+    ? decodeURIComponent(context.query.newEmailAddress as string)
+    : null;
+
+  async function handleRequest(body: PageBodyResponse, jwt: string) {
+    await addSchemeEditor(
+      schemeId,
+      jwt,
+      getUserTokenFromCookies(context.req),
+      body.editorEmailAddress
+    );
+    return body.editorEmailAddress;
+  }
```

Comment:
we may as well destructure this rather than accessing `body` twice
```suggestion
  async function handleRequest({ editorEmailAddress }: PageBodyResponse, jwt: string) {
    await addSchemeEditor(
      schemeId,
      jwt,
      getUserTokenFromCookies(context.req),
      editorEmailAddress
    );
    return editorEmailAddress;
  }
```